### ISSUE TITLE
docker: failed to register layer: Error processing tar file(exit status 1): lchown /usr/bin/etcd: invalid argument.

### ISSUE THREAD

**Issue by PBXg33k** [rancher/rancher#24910]
**What kind of request is this (question/bug/enhancement/feature request):** Bug

**Steps to reproduce (least amount of steps as possible):** 
simply running
```
docker run -d --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher
```

**Result:**
```
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
7ddbc47eeb70: Pull complete 
c1bbdc448b72: Pull complete 
8c3b70e39044: Pull complete 
45d437916d57: Pull complete 
28a9e619eb92: Pull complete 
3922b811dfe2: Pull complete 
81d9d43baeba: Pull complete 
ea183fe1d558: Pull complete 
59cb3488b9cd: Extracting [==================================================>]  99.63MB/99.63MB
470bfb078fde: Download complete 
9d3364413c7e: Download complete 
dcceab1151c5: Download complete 
22541110ae37: Download complete 
594323731b75: Download complete 
docker: failed to register layer: Error processing tar file(exit status 1): lchown /usr/bin/etcd: invalid argument.
See 'docker run --help'.
```

**Other details that may be helpful:**
Fresh install on a Proxmox server as a container using the debian-9.0 standard image.

Accidentally wiped my old rancher server's disk image and was in the process of recreating the container to restore/rebuild my rancher installation.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): rancher/rancher:latest
- Installation option (single install/HA): single install

- Docker version (use `docker version`):
```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:25:58 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:24:29 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683```

**Comment by liltimtim** [comment]
Similar issue running steps: 

```
docker run -d --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
5c939e3a4d10: Pull complete 
c63719cdbe7a: Pull complete 
19a861ea6baf: Pull complete 
651c9d2d6c4f: Pull complete 
6d1c86a401db: Pull complete 
c7d485afd256: Pull complete 
285d247a5c2b: Pull complete 
e7acc0299fc2: Pull complete 
7346f9f2da73: Extracting  99.62MB/99.62MB
91f1fe4c3d21: Download complete 
8b36bf060d06: Download complete 
1e46e34177f8: Download complete 
08aff7247104: Download complete 
bf9b23e67888: Download complete 
docker: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcd: invalid argument.
See 'docker run --help'.
```

**Comment by Napsty** [comment]
Similar here in a privileged LXC container using Ubuntu 18.04:

```
oot@ccp-ran01-u:~# docker run -d --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher:stable
Unable to find image 'rancher/rancher:stable' locally
stable: Pulling from rancher/rancher
5bed26d33875: Pull complete 
f11b29a9c730: Pull complete 
930bda195c84: Pull complete 
78bf9a5ad49e: Pull complete 
12a73929b6a7: Pull complete 
8434af3b0a23: Pull complete 
28db93a68de0: Pull complete 
e6dfd852f705: Pull complete 
a1fa824ccd2c: Extracting [==================================================>]  99.67MB/99.67MB
1e2d165916be: Download complete 
aaf1116b238c: Download complete 
375fded79e14: Download complete 
e2c84878ed8a: Download complete 
f7a8fcb48ebd: Download complete 
docker: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcd: invalid argument.
See 'docker run --help'.
```

**Comment by HalisCz** [comment]
Similar issue here on vpsfree.cz cloud (LXC based) with Ubuntu 18.04 and 20.04:

> root@rancher-test:~# docker pull rancher/rke-tools:v0.1.72
> v0.1.72: Pulling from rancher/rke-tools
> e95f33c60a64: Pull complete
> 6b9066ff94f0: Pull complete
> d00048cae6c8: Pull complete
> 673a80f76512: Pull complete
> 5265c6a8bcaa: Pull complete
> bd5a2b7ec0a7: Pull complete
> 0119010d361b: Pull complete
> 04670b023a98: Pull complete
> 96787bdffc36: Pull complete
> 11b333b25e38: Pull complete
> edec598acdb7: Extracting [==================================================>]  5.101MB/5.101MB
> c66bbb7a280c: Download complete
> 199d7f942076: Download complete
> f6eba55964bb: Download complete
> e868bdb5d818: Download complete
> cbec9cc5bcb9: Download complete
> failed to register layer: Error processing tar file(exit status 1): lchown /usr/local/bin/etcdctl: invalid argument

It seems that this image requires some really high uid to set (6810230), which is not available in my LXC (max available is 524287)

> fchownat(AT_FDCWD, "/usr/local/bin/etcdctl", 6810230, 100, AT_SYMLINK_NOFOLLOW <unfinished ...>
> <... fchownat resumed> )        = -1 EINVAL (Invalid argument)

**Comment by Genetical** [comment]
This issue still exists:
```
genetical@genetical:~$ docker run -d --restart=unless-stopped \
>   -p 80:80 -p 443:443 \
>   --privileged \
>   rancher/rancher:latest
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
a3009803982d: Pull complete
cf9e817c5d35: Pull complete
b380663f8ccc: Pull complete
1ca0e2238656: Pull complete
65086cb458c9: Pull complete
c6d25608690f: Pull complete
6c8ad6da7ce2: Pull complete
6a6940e66f68: Pull complete
b115b1ef2b5b: Pull complete
b4b03dbaa949: Extracting [==================================================>]  68.97MB/68.97MB
aef7deb59b77: Download complete
0bbf7579a568: Download complete
eaa5d6336f95: Download complete
608f536609b9: Download complete
fcaf65f7937c: Download complete
e4ea550002d9: Download complete
9d698b9289d2: Download complete
caa4144aedf1: Download complete
docker: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcdctl: invalid argument.
See 'docker run --help'.
```

**Comment by HammerHeston** [comment]
I'm getting the same results even just trying to pull the image without executing.

brian@ubuntu:~$ docker image pull rancher/rancher:latest
latest: Pulling from rancher/rancher
a3009803982d: Pull complete
cf9e817c5d35: Pull complete
b380663f8ccc: Pull complete
1ca0e2238656: Pull complete
65086cb458c9: Pull complete
c6d25608690f: Pull complete
6c8ad6da7ce2: Pull complete
6a6940e66f68: Pull complete
b115b1ef2b5b: Pull complete
b4b03dbaa949: Extracting [==================================================>]  68.97MB/68.97MB
aef7deb59b77: Download complete
0bbf7579a568: Download complete
eaa5d6336f95: Download complete
608f536609b9: Download complete
fcaf65f7937c: Download complete
e4ea550002d9: Download complete
9d698b9289d2: Download complete
caa4144aedf1: Download complete
failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcdctl: invalid argument

**Comment by mamyia** [comment]
Did sb managed to fox this? I have same problem at Proxmox CT in deb 
With out **sudo commad**
`mamyia@Docker ~$ docker run -d --restart=unless-stopped \
>   -p 80:80 -p 443:443 \
>   --privileged \
>   rancher/rancher:latest
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
a3009803982d: Pull complete 
cf9e817c5d35: Pull complete 
b380663f8ccc: Pull complete 
1ca0e2238656: Pull complete 
65086cb458c9: Pull complete 
c6d25608690f: Pull complete 
6c8ad6da7ce2: Pull complete 
6a6940e66f68: Pull complete 
b115b1ef2b5b: Pull complete 
b4b03dbaa949: Extracting [==================================================>]  68.97MB/68.97MB
aef7deb59b77: Download complete 
0bbf7579a568: Download complete 
eaa5d6336f95: Download complete 
608f536609b9: Download complete 
fcaf65f7937c: Download complete 
e4ea550002d9: Download complete 
9d698b9289d2: Download complete 
caa4144aedf1: Download complete 
docker: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcdctl: invalid argument.
See 'docker run --help'.
`
same result with **sudo**
`mamyia@Docker ~$ sudo docker run -d --restart=unless-stopped   -p 80:80 -p 443:443   --privileged   rancher/rancher:latest
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
a3009803982d: Pull complete 
cf9e817c5d35: Pull complete 
b380663f8ccc: Pull complete 
1ca0e2238656: Pull complete 
65086cb458c9: Pull complete 
c6d25608690f: Pull complete 
6c8ad6da7ce2: Pull complete 
6a6940e66f68: Pull complete 
b115b1ef2b5b: Pull complete 
b4b03dbaa949: Extracting [==================================================>]  68.97MB/68.97MB
aef7deb59b77: Download complete 
0bbf7579a568: Download complete 
eaa5d6336f95: Download complete 
608f536609b9: Download complete 
fcaf65f7937c: Download complete 
e4ea550002d9: Download complete 
9d698b9289d2: Download complete 
caa4144aedf1: Download complete 
docker: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcdctl: invalid argument.
See 'docker run --help'.
`
docker version:
`mamyia@Docker ~$ docker version
Client: Docker Engine - Community
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:47 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0`
debain version:

`mamyia@Docker ~$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
`

**Comment by somehume** [comment]
> I'm getting the same results even just trying to pull the image without executing.
> 
> brian@ubuntu:~$ docker image pull rancher/rancher:latest latest: Pulling from rancher/rancher a3009803982d: Pull complete cf9e817c5d35: Pull complete b380663f8ccc: Pull complete 1ca0e2238656: Pull complete 65086cb458c9: Pull complete c6d25608690f: Pull complete 6c8ad6da7ce2: Pull complete 6a6940e66f68: Pull complete b115b1ef2b5b: Pull complete b4b03dbaa949: Extracting [==================================================>] 68.97MB/68.97MB aef7deb59b77: Download complete 0bbf7579a568: Download complete eaa5d6336f95: Download complete 608f536609b9: Download complete fcaf65f7937c: Download complete e4ea550002d9: Download complete 9d698b9289d2: Download complete caa4144aedf1: Download complete failed to register layer: ApplyLayer exit status 1 stdout: stderr: lchown /usr/bin/etcdctl: invalid argument
> 
> But if I run with sudo, it works just fine. There appears to be a problem with it running "rootless".

This is the exact error that I have. Thanks, it did work with sudo... which I didn't consider because of the rootless thing.

**Comment by deven2728** [comment]
useing the command "sudo usermod -aG docker $USER && newgrp docker" instead of "dockerd-rootless-setuptool.sh install" fixed the problem for me. "dockerd-rootless-setuptool.sh install" seems to break it.

**Comment by mshavliuk** [comment]
Same issue happened with me as well. I used `dockerd-rootless-setuptool.sh` with the following `docker run` without `sudo` as suggested in [this guide](https://ranchermanager.docs.rancher.com/pages-for-subheaders/rancher-on-a-single-node-with-docker#option-a-default-rancher-generated-self-signed-certificate). After I ran the same command with sudo the issue was resolved.

**Comment by xamafe** [comment]
the new group didn't work for me:

`rancher@rancher:~$ groups
rancher cdrom floppy sudo audio dip video plugdev netdev docker
rancher@rancher:~$ docker run -d --restart=unless-stopped   -p 80:80 -p 443:443   -v /home/rancher/cert/fullchain.pem:/etc/rancher/ssl/cert.pem   -v /home/rancher/cert/privkey.pem:/etc/rancher/ssl/key.pem   -v /volumes/data:/var/lib/rancher   --privileged   rancher/rancher:latest   --no-cacerts
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
cac05f14fd97: Pull complete
560518f9ed79: Pull complete
d826c57abd3c: Pull complete
60a36fad79b4: Pull complete
2a5f788cf10e: Pull complete
ce9024822cf4: Pull complete
6618f6edc4eb: Pull complete
baf4a0ffb83a: Pull complete
9d8432399499: Pull complete
bd7c06ae6ecf: Pull complete
669801f22019: Pull complete
85a9b141881f: Extracting [==================================================>]   10.4MB/10.4MB
9b231c48f472: Download complete
7047f5709499: Download complete
9da445dcc1d1: Download complete
84ba23ff2968: Download complete
47cae4ad0c09: Download complete
3c6800130f2b: Download complete
f88bb00caa76: Download complete
0c8d48563d20: Download complete
docker: failed to register layer: ApplyLayer exit status 1 stdout:  stderr: lchown /usr/bin/etcdctl: invalid argument.
See 'docker run --help'.`

**Comment by adrlas** [comment]
> Similar issue here on vpsfree.cz cloud (LXC based) with Ubuntu 18.04 and 20.04:
> 
> > root@rancher-test:~# docker pull rancher/rke-tools:v0.1.72
> > v0.1.72: Pulling from rancher/rke-tools
> > e95f33c60a64: Pull complete
> > 6b9066ff94f0: Pull complete
> > d00048cae6c8: Pull complete
> > 673a80f76512: Pull complete
> > 5265c6a8bcaa: Pull complete
> > bd5a2b7ec0a7: Pull complete
> > 0119010d361b: Pull complete
> > 04670b023a98: Pull complete
> > 96787bdffc36: Pull complete
> > 11b333b25e38: Pull complete
> > edec598acdb7: Extracting [==================================================>]  5.101MB/5.101MB
> > c66bbb7a280c: Download complete
> > 199d7f942076: Download complete
> > f6eba55964bb: Download complete
> > e868bdb5d818: Download complete
> > cbec9cc5bcb9: Download complete
> > failed to register layer: Error processing tar file(exit status 1): lchown /usr/local/bin/etcdctl: invalid argument
> 
> It seems that this image requires some really high uid to set (6810230), which is not available in my LXC (max available is 524287)
> 
> > fchownat(AT_FDCWD, "/usr/local/bin/etcdctl", 6810230, 100, AT_SYMLINK_NOFOLLOW <unfinished ...>
> > <... fchownat resumed> )        = -1 EINVAL (Invalid argument)

I confirm that it is linked to a uid/gid generation issue.
I solved the problem (Proxmox 7.4 + CT ubuntu 22 + fuse-overlay) following the instructions from there : https://forum.proxmox.com/threads/how-to-use-uids-gids-higher-than-65535-in-ct-lxc.66382/
using uid and gid up to 8000000
